src/test/java/io/vertx/jdbcclient/JDBCPoolTest.java

```diff
@@ -379,4 +380,52 @@ public void testPreparedStatementWithBufferParam(TestContext should) {
         )
       );
   }
+  
+  @Test
+  public void testConnectionReturnedToPoolOnFailingQueryExecution(TestContext should) {
+    final Async test = should.async();
```

Comment:
how does the test checks that the connection was actually closed ?

Reply:
The superclass of `JDBCPoolTest` configures are pool with max size 1
https://github.com/vert-x3/vertx-jdbc-client/blob/3561a6fc52189f737d96a5e327c4816af1a464da/src/test/java/io/vertx/jdbcclient/ClientTestBase.java#L36-L38

In the test we are executing two failing queries. 
https://github.com/vert-x3/vertx-jdbc-client/blob/bd6f836c64a3c21207827db28b105a5bff7f081d/src/test/java/io/vertx/jdbcclient/JDBCPoolTest.java#L390-L404
The first one will fail with a `SQLSyntaxErrorException`. If the according connection is closed and returned to the pool, the second query will fail immediately with the same `SQLSyntaxErrorException`. If the according connection is not closed, the pool will be exhausted and we will see an according `SQLException` after acquisition timeout.
```
java.sql.SQLException: Sorry, acquisition timeout!
	at io.agroal.pool.ConnectionPool.handlerFromSharedCache(ConnectionPool.java:304)
	at io.agroal.pool.ConnectionPool.getConnection(ConnectionPool.java:241)
	at io.agroal.pool.DataSource.getConnection(DataSource.java:81)
	at io.vertx.ext.jdbc.impl.JDBCClientImpl.lambda$5(JDBCClientImpl.java:226)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Reply:
thanks, can you add a comment in the test for this ? I think it's worth to be captured.

Reply:
Done.